server/jetstream_cluster.go

```diff
@@ -3375,6 +3385,30 @@ func (js *jetStream) processStreamAssignmentResults(sub *subscription, c *client
 
 	// FIXME(dlc) - suppress duplicates?
 	if sa := js.streamAssignment(result.Account, result.Stream); sa != nil {
+		canDelete := !result.Update && time.Since(sa.Created) < 5*time.Second
+
+		// See if we should retry in case this cluster is full but there are others.
+		if cfg, ci := sa.Config, sa.Client; cfg != nil && ci != nil && isInsufficientResourcesErr(result.Response.Error) && canDelete {
+			// If cluster is defined we can not retry.
+			if cfg.Placement == nil || cfg.Placement.Cluster == _EMPTY_ {
+				// If we have additional clusters to try we can retry.
+				if ci != nil && len(ci.Alternates) > 0 {
+					if rg := js.createGroupForStream(ci, cfg); rg != nil {
+						s.Warnf("Retrying cluster placement for stream '%s > %s'", result.Account, result.Stream)
```

Comment:
we should probably increment some stat counter when this happens and maybe send an advisory, and maybe clarify why this being retried here rather than just saying its being retried.  And include where the placement is being done instead in the log (and advisory)

Reply:
ok, maybe just some extra info in the log then to indicate its this, a user who come across this for the first time would not have any idea what it relates to

Reply:
Possibly if they were doing no monitoring at all of their system to see it ran out of resources. What would you add to the description that you think would be helpful in a system like NGS? Or any super cluster tbh.

Reply:
```Retrying cluster placement for stream 'X > Y'` due to insufficient resources in cluster C``?

Reply:
That works, thanks.

Reply:
Note though that this is the not happy path where the meta-leader guesses wrong around placement, either due to concurrent requests, dropped updates around reserved or real usage etc. 

In the happy path we have no logging at all atm.